numpy/core/src/multiarray/dtypemeta.h

```diff
@@ -1,7 +1,65 @@
 #ifndef _NPY_DTYPEMETA_H
 #define _NPY_DTYPEMETA_H
 
+
+/* DType flags, currently private, since we may just expose functions */
+#define NPY_DT_LEGACY 1 << 0
+#define NPY_DT_ABSTRACT 1 << 1
+#define NPY_DT_PARAMETRIC 1 << 2
+
+
+typedef struct {
+    /* DType methods, these could be moved into its own struct */
```

Comment:
Ah, clearly you were thinking about these as a separate entity as well...

Reply:
Yeah, the split here is more: definitely private/opaque struct, vs. not-so-private.  Splitting it out into multiple structs seems however just unnecessary.  I could keep these on the DType itself for now, but this has the nice advantage of being "truly" private.

Reply:
OK, that makes sense. Regardless, with the macros it becomes an implementation detail, which is nice.